cmd/gateway-azure.go

```diff
@@ -181,6 +156,65 @@ func azureToObjectError(err error, params ...string) error {
 	return e
 }
 
+func mustGetAzureUploadID() string {
+	var id [8]byte
+
+	n, err := io.ReadFull(rand.Reader, id[:])
+	fatalIf(err, "unable to generate upload ID for azure")
+	if n != len(id) {
+		fatalIf(fmt.Errorf("insufficient random data (expected: %d, read: %d)", len(id), n),
+			"unable to generate upload ID for azure")
+	}
+
+	return fmt.Sprintf("%x", id[:])
+}
+
+func checkAzureUploadID(uploadID string) (err error) {
```

Comment:
We would need some unit tests for these functions..

Reply:
done